app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -259,7 +259,7 @@ def save_network_ports_inventory(ems, hashes, target = nil)
         h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
       end
 
-      h[:security_groups] = h.fetch_path(:security_groups).map { |x| x[:_object] } if h.fetch_path(:security_groups, 0, :_object)
+      h[:security_groups] = (h.fetch_path(:security_groups) || []).map { |x| x.try(:[], :_object) }.compact.uniq
```

Comment:
Is the `.try` really necessary?  Is it possible for `h[:security_groups]` to contain `nil`s (as opposed to actually being `nil`)?